tree/dataframe/src/RNTupleDS.cxx

```diff
@@ -79,7 +79,7 @@ class RRDFCardinalityField final : public ROOT::RFieldBase {
       return representations;
    }
    // Field is only used for reading
-   void GenerateColumns() final { assert(false && "Cardinality fields must only be used for reading"); }
+   void GenerateColumns() final { throw RException(R__FAIL("Cardinality fields must only be used for reading")); }
```

Comment:
Since this is an internal field and defined only in the implementation, I'm wondering whether in this case it actually makes more sense to keep it as an assertation :thinking:, same for `RArraySizeField`.

Reply:
I tend to agree. Since technically these might be called by a user we might use `R__ASSERT` to make sure they hit it even in release builds (unless they disable assertions, in which case they would just get a no-op which is not catastrophic)

Reply:
That's fair enough